### ISSUE TITLE
Chances of getting a segmentation fault at address 0xFFFFFFFFFFFFFFFF when opening a websocket (connecting to MongoDB and Discord)

### ISSUE THREAD

**Issue by Sayrix** [oven-sh/bun#10695]
### How can we reproduce the crash?

`Windows 11 23H2 22631.3527`
`Bun 1.1.6`
`discord.js@14.14.1`
`mongodb@6.5.0`

```js
import * as Discord from "discord.js";
import { Db, MongoClient, ServerApiVersion } from "mongodb";

const client = new Discord.Client({
	intents: ["Guilds"],
	shards: "auto",
}) as Client;
const cache = new NodeCache();
const mongodb = new MongoClient(
	"mongodb+srv://URL_REDACTED/?retryWrites=true&w=majority",
	{
		serverApi: {
			version: ServerApiVersion.v1,
			strict: true,
			deprecationErrors: true,
		},
	}
);

mongodb.connect();
client.login(process.env.DISCORD_TOKEN as string);
```

Video of me getting the error, and sometimes not.

https://github.com/oven-sh/bun/assets/43046854/b94837f1-ee2c-49b3-b130-a6549362ed13

### Stack Trace (bun.report)

Bun v1.1.6 ([`e58d67b`](https://github.com/oven-sh/bun/tree/e58d67b46875280bc9002add4c3fae45472cb346)) on windows x86_64 [AutoCommand]

Segmentation fault at address 0xFFFFFFFFFFFFFFFF

- `??`
- `??`
- `quote_cmd_arg`
- `fs__open`
- `??`
- [`bun_js.zig:321`](https://github.com/oven-sh/bun/blob/e58d67b46875280bc9002add4c3fae45472cb346/src/bun_js.zig#L321): `start`
- `JSC__JSValue__callCustomInspectFunction`
- [`shimmer.zig:186`](https://github.com/oven-sh/bun/blob/e58d67b46875280bc9002add4c3fae45472cb346/src/bun.js/bindings/shimmer.zig#L186): `cppFn`
- [`run_command.zig:1300`](https://github.com/oven-sh/bun/blob/e58d67b46875280bc9002add4c3fae45472cb346/src/cli/run_command.zig#L1300): `exec`
- [`cli.zig:1852`](https://github.com/oven-sh/bun/blob/e58d67b46875280bc9002add4c3fae45472cb346/src/cli.zig#L1852): `start`

**Comment by Sayrix** [comment]
The bug is still here in 1.1.7

Bun v1.1.7 ([`b0b7db5`](https://github.com/oven-sh/bun/tree/b0b7db5c060e830801b0273b7ff6e34e3eb5c254)) on windows x86_64 [AutoCommand]

Segmentation fault at address 0xFFFFFFFFFFFFFFFF

- `??`
- `??`
- `uv__process_poll_req`
- `??`
- `uv_run`
- [`uws.zig:2619`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/deps/uws.zig#L2619): `run`
- `JSC__VM__holdAPILock`
- [`shimmer.zig:186`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/bun.js/bindings/shimmer.zig#L186): `cppFn`
- [`run_command.zig:1307`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/cli/run_command.zig#L1307): `exec`
- [`cli.zig:1861`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/cli.zig#L1861): `start`

```Bun v1.1.7 (b0b7db5c) Windows x64
Args: "C:\Users\Sayrix\.bun\bin\bun.exe", "."
Features: jsc Bun.stdin(2) dotenv transpiler_cache(5) tsconfig(8) WebSocket(3) 
Builtins: "bun:main" "node:buffer" "node:crypto" "node:dns" "node:events" "node:fs" "node:fs/promises" "node:http" "node:net" "node:os" "node:path" "node:stream" "node:string_decoder" "node:timers" "node:timers/promises" "node:tls" "node:url" "node:util" "node:util/types" "node:zlib" "node:worker_threads" "undici" "ws" 
Elapsed: 5455ms | User: 78ms | Sys: 15ms
RSS: 0.26GB | Peak: 0.26GB | Commit: 0.32GB | Faults: 62872

panic(main thread): Segmentation fault at address 0xFFFFFFFFFFFFFFFF
oh no: Bun has crashed. This indicates a bug in Bun, not your code.```

**Comment by Sayrix** [comment]
Hi, i stopped getting the error on Bun 1.1.17 Windows 11 23H2 22631.3737 x64 but now i have it on Bun 1.1.17 Ubuntu 22.04.4 LTS aarch64 

### Stack Trace (bun.report)

Bun v1.1.17 ([`bb66bba`](<https://github.com/oven-sh/bun/tree/bb66bba1bf6d57983e7efef7ab706bce12137028>)) on linux aarch64 [AutoCommand]

Segmentation fault at address 0x0000001A

- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `WTF::jscSignalHandler`
- [`process_common.rs:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/sys/pal/unix/process/process_common.rs#L0>): `??`
- *2 unknown/js code*
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `llint_call_javascript`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `JSC::Interpreter::executeCall`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `JSC::runJSMicrotask`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `JSC::VM::drainMicrotasks`
- [`BunProcess.cpp:2558`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/bun.js/bindings/BunProcess.cpp#L2558>): `Bun::jsFunctionDrainMicrotaskQueue`
- *1 unknown/js code*

Features: jsc, Bun.stdin, fetch, http\_server, transpiler\_cache, tsconfig\_paths, tsconfig, WebSocket

**Comment by Sayrix** [comment]
Just to let you know that the error is still here on Bun 1.1.21

Bun v1.1.21 ([`70ca2b7`](<https://github.com/oven-sh/bun/tree/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b>)) on linux aarch64 [AutoCommand]

Segmentation fault at address 0x0000001A

- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `WTF::jscSignalHandler`
- [`alloc.rs:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/src/alloc.rs#L0>): `??`
- *2 unknown/js code*
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `llint_call_javascript`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `JSC::Interpreter::executeCall`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `JSC::runJSMicrotask`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `JSC::VM::drainMicrotasks`
- [`BunProcess.cpp:2570`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/src/bun.js/bindings/BunProcess.cpp#L2570>): `Bun::jsFunctionDrainMicrotaskQueue`
- *1 unknown/js code*

Features: jsc, Bun.stdin, fetch, http\_server, transpiler\_cache, tsconfig\_paths, tsconfig, WebSocket

**Comment by cirospaciari** [comment]
Fixed in canary `v1.1.22`
